provisioning/ubuntu-provision.sh

```diff
@@ -198,6 +198,23 @@ function install_docker() {
   usermod -aG docker "${username}"
 }
 
+## Ensure Golang is installed
+function install_golang(){
+  local temp_archive="/tmp/go${GOLANG_VERSION}.tgz"
+  local installation_dir="/usr/local/"
+  local download_url="https://go.dev/dl/go${GOLANG_VERSION}.linux-${ARCHITECTURE}.tar.gz"
+  curl --fail --silent --show-error --location --output "${temp_archive}" "${download_url}"
+  tar --extract --gunzip --file="${temp_archive}" --directory="${installation_dir}"
+  # Symlink to a folder in the PATH
+  ln -s "${installation_dir}go/bin/go" /usr/local/bin/go
```

Comment:
Are there other files in the downloaded archive? If not or if these files aren't needed, I would extract only the go binary, directly to /usr/local/bin/go so the symbolic link wouldn't be needed anymore.